### PR TITLE
HAUKI-582: Add the option to specify which date periods to copy from a resource

### PR DESCRIPTION
**Pull request by danipran**
Add a new query parameter `date_period_ids` in the `/v1/resource/{id}/copy_date_periods/` endpoint, which accepts a comma-separated list of date period ids for specifying which date periods to copy to the target resource.

Also did some refactoring on the `copy_date_periods` tests.

### CODE REVIEW

File: hours/viewsets.py [hunk 1]
```diff
@@ -599,11 +606,36 @@ def copy_date_periods(self, request, pk=None):
             )
             raise PermissionDenied(detail=detail)
 
+        if date_period_ids := request.query_params.get("date_period_ids"):
```
Comment (nicobav) [inline hours/viewsets.py:609]: Look, a walrus! :=

File: hours/models.py [hunk 2]
```diff
@@ -547,15 +547,29 @@ def get_descendants(self, acc=None):
...
+        If replace is True, the target resource's existing date periods are deleted.
+        """
         if (
             not target_resource
             or not isinstance(target_resource, Resource)
             or self.id == target_resource.id
         ):
             return
 
+        date_period_ids = date_period_ids or []
```
Comment (nicobav) [inline hours/models.py:570]: I think we can get rid off this line

File: hours/models.py [hunk 3]
```diff
@@ -547,15 +547,29 @@ def get_descendants(self, acc=None):
...
         existing_period_ids = []
+
```
Comment (nicobav) [inline hours/models.py:572]: Ahh, some air, very nice👍

**Review by nicobav — APPROVED** [review]
LGTM 👍 

(left one cleaning suggestion 🧹 )

File: hours/models.py (hunk 3 above)
Comment (danipran) [reply hours/models.py:572]: A fellow man of culture, I see 🧐

File: hours/models.py (hunk 2 above)
Comment (danipran) [reply hours/models.py:570]: Good point!

File: hours/viewsets.py (hunk 1 above)
Comment (danipran) [reply hours/viewsets.py:609]: My shiny new toy 🤩

**Pull request merged**